### ISSUE TITLE
Node naming inconsistency in new Revit getter nodes

### ISSUE THREAD

**Issue by andydandy74** [DynamoDS/DynamoRevit#1663]
## Dynamo version

1.3 stable

## Revit version

Any

## Operating system

Any

## What did you do? 

Had a look at some of the new Revit nodes

## What did you expect to see?

Node names that follow the Dynamo naming scheme

## What did you see instead?

Node names that *don't* follow the Dynamo naming scheme, e.g. ```Element.GetLocation``` should be named ```Element.Location``` etc.
Until Dynamo 1.2, all nodes that retrieve element properties were named ```Element.{PROPERTYNAME}```, not ```Element.Get{PROPERTYNAME}``` - I think the original naming scheme should be adhered to.

**Comment by ksobon** [comment]
@andydandy74 did we miss any more of these?

**Comment by eibre** [comment]
@andydandy74 What about the good old Element.GetParameterValueByName? Is it the only exception?

**Comment by andydandy74** [comment]
@ksobon - These are the ones I had found:
- Element.GetCategory
- Element.GetLocation
- Element.GetMaterials
- FamilyInstance.GetFamily
- FamilyInstance.GetHost
- FamilyInstance.GetType
- FillPatternElement.GetByName
- LinePatternElement.GetByName

@eibre - Element.GetParameterValueByName is technically not a Property node but an Actions node in that it expects a second argument. Also, I think people have gotten used to this node's name by now whereas all the other ones are fairly new. Then again, if we look at the very first Dynamo node naming rule, it shouldn't be an exception:
https://github.com/DynamoDS/Dynamo/wiki/Naming-Standards#node-names

**Comment by ksobon** [comment]
Well @andydandy74 I would love to go after these, but I need a little example from @mjkkirschner or someone on the dev team on how to handle migrations for ZT nodes. Changing a name of the node, will break old definitions so we need to address that...

**Comment by eibre** [comment]
@andydandy74 from the naming standard:

>Use parallel structures for like functionalities.
>
>Ex: Element.SetParameter and >Element.GetParameter

**Comment by andydandy74** [comment]
@eibre yeah I saw that one, too. Decided to go with the first rule, though... ;-)

**Comment by ksobon** [comment]
I would say something like `Location` does make sense to do Get/Set because you can do both with the same/similar API calls. Other things like `Category` is read only and to set a category on a family is a totally different process. Again, get/set for parameters makes sense, but not for `getType` because the Type gets set via parameter....so on and so on.

**Comment by andydandy74** [comment]
Sure about that? If we follow that logic we can start renaming all the XXX.Name nodes to XXX.GetName as well... ;-)

**Comment by ksobon** [comment]
That should all be handled in a single `Element.Name` node...I am not sure why we even have all of these nodes in there...and yes, I know that some element types won't return a name that way, but majority does, and we can handle the rest internally....i think.

**Comment by eibre** [comment]
Interesting.. there are twelve *.Name nodes and only one .SetName node (Room.SetName). I totally agree that _one_ Element.Name and _one_ Element.SetName should be enough.

**Comment by eibre** [comment]
By the way, some of the new nodes are in the wrong subcategories as well:
![image](https://cloud.githubusercontent.com/assets/5968613/26287586/197ee00e-3e7f-11e7-9230-e240c15790a8.png)

**Comment by smangarole** [comment]
@Racel ping.

**Comment by Racel** [comment]
@riteshchandawar @sharadkjaiswal @Randy-Ma - As we are going through the library re-categorization work, can we make some of these adjustments?

- Can we get the nodes under the right node type (create, action, query)?
- Can we evaluate the Get/Set nodes @andydandy74 reviewed above to see which ones are essential and determine the correct naming? 

Let me know. Reassigning to @riteshchandawar for now.

**Comment by riteshchandawar** [comment]
@Racel Right now we are not considering the recategorization of **Revit** nodes, we have done that for everything else including Geometry Library.

Let's talk on email and discuss further on Revit nodes.

- R

**Comment by Racel** [comment]
Thanks @riteshchandawar
